src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -323,6 +393,33 @@ protected override void Apply(TransactionOperationContext context, BlittableJson
             }
         }
 
+        private void SetIndexForBackup(TransactionOperationContext toc, BlittableJsonReaderObject bjro, long index, string type)
+        {
+            if (index < 0)
+                return;
+
+            if ((bjro.TryGet(nameof(CompareExchangeCommandBase.DatabaseName), out string databaseName) == false ||
+                 string.IsNullOrEmpty(databaseName)) && type.Equals(nameof(AddOrUpdateCompareExchangeCommand)))
+                throw new RachisApplyException("Update database command must contain a DatabaseName property");
+
+            var dbKey = $"db/{databaseName}";
+            var items = toc.Transaction.InnerTransaction.OpenTable(ItemsSchema, Items);
+
+            using (Slice.From(toc.Allocator, dbKey, out Slice key))
+            using (Slice.From(toc.Allocator, dbKey.ToLowerInvariant(), out Slice keyLowered))
+            {
+                var databaseRecordJson = ReadInternal(toc, out _, keyLowered);
+                if (databaseRecordJson == null)
+                    return;
+
+                var databaseRecord = JsonDeserializationCluster.DatabaseRecord(databaseRecordJson);
```

Comment:
You can modify the blittable directly. No need to deserialzie it.

Reply:
done